Silence sysctl read of LDM state in sandbox

pvollan opened the pull request:
#### 4a7e72df8752c9c45c36de66a55519936da6482a
<pre>
Silence sysctl read of LDM state in sandbox
<a href="https://bugs.webkit.org/show_bug.cgi?id=257263">https://bugs.webkit.org/show_bug.cgi?id=257263</a>
rdar://109737984

Reviewed by Brent Fulgham.

Silence sysctl read of LDM state in the WebContent process sandbox.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: <a href="https://commits.webkit.org/264490@main">https://commits.webkit.org/264490@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/15a4ea023f3fe686a341de87e37b72ba5e39c899

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/7804 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/8079 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/8261 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/9448 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/7946 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/10070 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/7996 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/10809 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/7942 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/9048 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/7151 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/9560 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/6357 "Passed tests") | | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/14758 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/7497 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/7237 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/10631 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/7726 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/6293 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/7048 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/1860 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/11258 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/7459 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -709,6 +709,7 @@
         "kern.osversion"
         "kern.secure_kernel" ;; Needed by XPC bundle resolution
         "kern.version"
+        "security.mac.lockdown_mode_state"
```
It seems like we would need this in the GPU and Network process, too.

brentfulgham reviewed (changes requested):
I do not think we should make this change, as I believe it will break the ability for Safari to exclude some sites from LDM behavior.

brentfulgham commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -716,16 +716,15 @@
 )
 
-(deny sysctl-read (with no-report)
-    (sysctl-name "vm.task_no_footprint_for_debug"))
```
Did you intend to remove this?

pvollan replied:
Yes, this will now be covered by the generic deny case, so we don't need a specific rule.

pvollan replied:
Yes, this will now be covered by the generic deny case, so we don't need a specific rule.

> Looks good -- thank you for answering my questions! :-)

NP, thanks for the approval!

brentfulgham commented on Tools/TestWebKitAPI/Tests/WebKitCocoa/Proxy.mm:
```diff
@@ -24,7 +24,7 @@
 
 #import "config.h"
-
+#if 0
```
Oops! I don't think you meant to include this!

pvollan replied:
Oops, fixed!

brentfulgham commented on Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
```diff
@@ -663,16 +663,14 @@
 )
 
-(deny sysctl-read (with no-report)
-    (sysctl-name "vm.task_no_footprint_for_debug"))
```
Ditto! Was this intended?

brentfulgham commented on Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
```diff
@@ -716,16 +716,15 @@
 (allow sysctl-read (sysctl-name "vm.malloc_ranges")) ;; <rdar://problem/105161083>
 #endif
 
+(deny sysctl-write (with no-report) (with telemetry))
```
This doesn't seem relevant to this particular bug report. Was this new telemetry you wanted to capture for another bug?

pvollan replied:
This was an intentional change, since I think it would be good if we could inspect telemetry for which sysctls are being denied without having local reports.

Thanks for reviewing!

brentfulgham reviewed (approved):
Looks good -- thank you for answering my questions! :-)

pvollan committed:
Silence sysctl read of LDM state in sandbox
https://bugs.webkit.org/show_bug.cgi?id=257263
rdar://109737984

Reviewed by Brent Fulgham.

Silence sysctl read of LDM state in the WebContent process sandbox.

* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:

Canonical link: https://commits.webkit.org/264490@main
